docs/streams/upgrade-guide.html

```diff
@@ -161,6 +177,65 @@ <h3><a id="streams_api_changes_400" href="#streams_api_changes_400">Streams API
         <code>TransformerSupplier</code>, <code>ValueTransformer</code>, and <code>ValueTransformerSupplier</code>.
     </p>
 
+    <p>
+        Kafka Streams introduced the <code>RETRY</code> return option the <code>ProductionExceptionHandler</code> (<a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=311627309">KIP-1065</a>).
+        This feature addresses the issue of infinite retry loops caused by <code>TimeoutException</code> when an output topic might be missing.
+        Now the application is enabled to attempt the action again instead of either crashing or dropping records.
+        This change aims to enhance the robustness and reliability of Kafka Streams by providing a more flexible error-handling mechanism.
+    </p>
+
+    <p>
+
+    </p>
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-1076%3A++Metrics+for+client+applications+KIP-714+extension">KIP-1076</a>
+        proposes adding methods to the Admin, Consumer, and Producer interfaces to allow applications embedding Kafka clients to register and push application-specific metrics,
+        supplementing the client metrics introduced in KIP-714.
```

Comment:
> the client metrics introduced in

KIP-714 did not introduce new metrics IIRC (and even if, it would not be the point, right?) -- the KIP is about collection client metrics broker side.